crates/rattler_repodata_gateway/src/gateway/query.rs

```diff
@@ -130,33 +140,57 @@ impl GatewayQuery {
             }
         }
 
-        // A list of futures to fetch the records for the pending package names. The main task
-        // awaits these futures.
+        let len = subdirs.len();
+        let mut result = vec![RepoData::default(); len];
```

Comment:
Should we add a special entry for direct urls? 

I think right now you are just adding them to the results of the first subdir .

Reply:
I don't know what depends on these subdirs. Does it matter for the result?

Reply:
https://github.com/mamba-org/rattler/pull/725/commits/c1266936f12563d65dcef5e6b0eefd543284eb2 I added a subdir here.